packages/kit/src/core/postbuild/prerender.js

```diff
@@ -117,7 +117,10 @@ async function prerender({ hash, out, manifest_path, metadata, verbose, env }) {
 		return { prerendered, prerender_map };
 	}
 
-	const emulator = await config.adapter?.emulate?.();
+	const emulator = await config.adapter?.emulate?.({
+		importEntryPoint: (entry) =>
+			import(pathToFileURL(`${config.outDir}/output/server/adapter/${entry}.js`).href)
+	});
```

Comment:
it looks like `emulator.beforeRequest` never gets called during prerender?

Reply:
As discussed: It's primary use case is to emulate things like middleware during dev/preview, and during prerendering there's no e.g. edge middleware involved, so it should not run.